File: Diploma.Server/Models/ExpertEvaluation.cs
using System.ComponentModel.DataAnnotations;

namespace Diploma.Server.Models
{
    public class ExpertEvaluation
    {
        [Key]
        public int EvaluationId { get; set; }
        public required string ProductId { get; set; }
        public required Product Product { get; set; }
        public required int ExpertId { get; set; }
        public required LLMExpert Expert {  get; set; }
        public double PriceStrategy { get; set; }
        public double Demand { get; set; }
        public double Quality { get; set; }
        public double PriceQuality { get; set; }
        public DateTime Timestamp { get; set; }
        public string[]? Advantages { get; set; }
        public string[]? Disadvantages { get; set; }
    }
}

File: diploma.client/src/models/ExpertEvaluation.ts
import LLMExpert from "./LLMExpert";
import Product from "./Product";

export default class ExpertEvaluation {
    evaluationId: number;
    productId: string;
    product: Product;
    expertId: number;
    expert: LLMExpert;
    priceStrategy: number;
    demand: number;
    quality: number;
    priceQuality: number;
    timestamp: Date;
    advantages: string[];
    disadvantages: string[];

    constructor() {
        this.evaluationId = 0;
        this.productId = "";
        this.product = new Product();
        this.expertId = 0;
        this.expert = new LLMExpert();
        this.priceStrategy = 0;
        this.demand = 0;
        this.quality = 0;
        this.priceQuality = 0;
        this.timestamp = new Date();
        this.advantages = [];
        this.disadvantages = [];
    }
}